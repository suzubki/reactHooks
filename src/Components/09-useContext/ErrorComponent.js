import React from 'react'

export const ErrorComponent = () => {
  return (
    <div>ErrorComponent</div>
  )
}
