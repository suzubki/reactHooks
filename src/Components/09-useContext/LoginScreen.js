import React, { useContext } from 'react'
import { UserContext } from './userContext'

export const LoginScreen = () => {

  const { setUser } = useContext( UserContext )

  const newUser = {
    id:'123',
    name: 'darwin'
  }

  return (
    <div>
        <h1>LoginScreen </h1>
        <hr/>

        <button
          className='btn btn-primary'
          onClick={ () => setUser({...newUser})}
        >
          Establecer nuevo valor
        </button>

    </div>
  )
}
