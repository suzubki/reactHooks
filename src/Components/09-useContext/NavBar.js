import React from 'react'
import { NavLink } from "react-router-dom";

export const NavBar = () => {
  return (
    <nav className="navbar navbar-expand-lg navbar-dark bg-dark">
      <div className="container-fluid">
        <div className="collapse navbar-collapse" id="navbarNav">
          <ul className="navbar-nav">
            <li className='nav-item'>
              <NavLink className='nav-link text-white' aria-current="page" to='/'>HomeScreen</NavLink>
            </li>
            <li className='nav-item'>
              <NavLink className='nav-link text-white' to='/about'>AboutScreen</NavLink>
            </li>
            <li className='nav-item'>
              <NavLink className='nav-link text-white' to='/login'>LoginScreen</NavLink>
            </li>
          </ul>
        </div>
      </div>
    </nav>

  )
}
