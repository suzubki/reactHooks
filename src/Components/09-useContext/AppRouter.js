import React from 'react'
import {
    BrowserRouter,
    Routes,
    Route,
} from "react-router-dom";
import { AboutScreen } from './AboutScreen';
import { LoginScreen } from './LoginScreen';
import { HomeScreen } from './HomeScreen';
import { ErrorComponent } from './ErrorComponent';
import { NavBar } from './NavBar';

export const AppRouter = () => {
    return (
        <div>
        <BrowserRouter>
            <NavBar />
            <Routes>    
                <Route path="/" element={<HomeScreen />} />

                <Route path="/about" element={<AboutScreen />} />
                <Route path="/login" element={<LoginScreen />} />
                <Route path='*' element={<ErrorComponent />} />
            </Routes>
        </BrowserRouter>
        </div>
  )
}
