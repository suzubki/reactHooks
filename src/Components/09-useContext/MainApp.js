import React, { useState } from 'react'
import { AppRouter } from './AppRouter'
import { UserContext } from './userContext'

export const MainApp = () => {

    const [user, setUser] = useState({})

    return (
        <div>
            <UserContext.Provider value={ {
                user,
                setUser
            } } >
                <AppRouter />
            </UserContext.Provider>
        </div>
    )
}
