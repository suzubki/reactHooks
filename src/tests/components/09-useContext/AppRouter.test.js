import { AppRouter } from "../../../Components/09-useContext/AppRouter";
import { UserContext } from "../../../Components/09-useContext/userContext";
import { mount } from "enzyme";

describe("<AppRouter />", () => {
    const user = {
        name: "user",
        email: "user@example.com",
    };

    const wrapper = mount(
        <UserContext.Provider value={{ user }}>
            <AppRouter />
        </UserContext.Provider>
    );

    test("should disply", () => {
        expect(wrapper).toMatchSnapshot();
    });
});
