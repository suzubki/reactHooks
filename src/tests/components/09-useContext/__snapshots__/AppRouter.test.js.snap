// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppRouter /> should disply 1`] = `
<div>
  Array [
    <nav
      className="navbar navbar-expand-lg navbar-dark bg-dark"
    >
      <div
        className="container-fluid"
      >
        <div
          className="collapse navbar-collapse"
          id="navbarNav"
        >
          <ul
            className="navbar-nav"
          >
            <li
              className="nav-item"
            >
              <a
                aria-current="page"
                className="nav-link text-white active"
                href="/"
                onClick={[Function]}
              >
                HomeScreen
              </a>
            </li>
            <li
              className="nav-item"
            >
              <a
                className="nav-link text-white"
                href="/about"
                onClick={[Function]}
              >
                AboutScreen
              </a>
            </li>
            <li
              className="nav-item"
            >
              <a
                className="nav-link text-white"
                href="/login"
                onClick={[Function]}
              >
                LoginScreen
              </a>
            </li>
          </ul>
        </div>
      </div>
    </nav>,
    <div>
      <h1>
        HomeScreen 
      </h1>
      <hr />
      <pre>
        {
   "name": "user",
   "email": "user@example.com"
}
      </pre>
    </div>,
  ]
</div>
`;
